hackathon_site/dashboard/frontend/src/components/general/Header/Header.js

```diff
@@ -5,39 +5,71 @@ import AppBar from "@material-ui/core/AppBar";
+import Drawer from "@material-ui/core/Drawer";
 import Hidden from "@material-ui/core/Hidden";
 import Menu from "@material-ui/icons/Menu";
-import Navbar from "components/general/Navbar/Navbar";
+import { Navbar, NavbarDrawer } from "components/general/Navbar/Navbar";
+import { userEmail, cartQuantity } from "testing/mockData";
 
-const Header = ({ showNavbar = true }) => (
-    <AppBar className={styles.header} position="sticky">
-        <Toolbar className={styles.headerToolbar}>
-            <Hidden implementation="css" smDown={!!showNavbar}>
-                <div className={styles.headerLogo}>
-                    <Logo
-                        className={styles.headerLogoImg}
-                        alt="Hackathon logo"
-                        data-testid="headerLogo"
-                    />
-                    <Typography variant="h6" data-testid="hackathonName">
-                        Hackathon Name
-                    </Typography>
-                </div>
-            </Hidden>
-            <Hidden implementation="css" mdUp>
-                {showNavbar && (
-                    <IconButton
-                        color="inherit"
-                        aria-label="Menu"
-                        className={styles.headerIconBtn}
-                    >
-                        <Menu />
-                    </IconButton>
-                )}
-            </Hidden>
-            {showNavbar && <Navbar />}
-        </Toolbar>
-    </AppBar>
+const HackathonTitle = () => (
+    <>
+        <Logo
+            className={styles.headerLogoImg}
+            alt="Hackathon logo"
+            data-testid="headerLogo"
+        />
+        <Typography variant="h6" data-testid="hackathonName">
+            Hackathon Name
+        </Typography>
+    </>
 );
 
+const Header = ({ showNavbar = true }) => {
+    const [mobileOpen, setMobileOpen] = React.useState(false);
+
+    const toggleMenu = () => {
+        setMobileOpen(!mobileOpen);
+    };
+
+    return (
+        <AppBar className={styles.header} position="sticky">
+            <Toolbar className={styles.headerToolbar}>
+                <Hidden implementation="css" smDown={!!showNavbar}>
+                    <div className={styles.headerLogo}>
+                        <HackathonTitle />
+                    </div>
+                </Hidden>
+                <Hidden implementation="css" mdUp>
+                    {showNavbar && (
+                        <IconButton
+                            color="inherit"
+                            aria-label="Menu"
+                            className={styles.headerIconBtn}
+                            onClick={toggleMenu}
+                            data-testid="menuBtn"
+                        >
+                            <Menu />
+                        </IconButton>
+                    )}
+                </Hidden>
+                {showNavbar && <Navbar cartQuantity={cartQuantity} />}
```

Comment:
This has it's own `Hidden` inside. Can we take advantage of the ones in this outer component instead?